casbin/internal_api.cpp

```diff
@@ -65,8 +65,11 @@ bool Enforcer :: addPolicies(string sec, string p_type, vector<vector<string>> r
         this->BuildIncrementalRoleLinks(policy_add, p_type, rules);
 
     if (this->adapter != NULL && this->auto_save) {
-        void* adapter = this->adapter;
-        ((BatchAdapter *)adapter)->AddPolicies(sec, p_type, rules);
+        try {
+            dynamic_cast<BatchAdapter*>(this->adapter)->AddPolicies(sec, p_type, rules);
+        }
+        catch(UnsupportedOperationException e) {
```

Comment:
Do we need to do anything in catch?

Reply:
No nothing, just we need to log the message, but casbin-cpp logger has not yet developed, so for now, we leave the catch block empty.